example/example.go

```diff
@@ -19,23 +21,32 @@ const (
 	region     = "us-east-2"
 )
 
-func run() error {
-	streamurl := fedoracoreos.GetStreamURL(fedoracoreos.StreamStable)
-	resp, err := http.Get(streamurl.String())
+func downloadISO(fcosstable stream.Stream) error {
+	iso := fcosstable.Architectures[targetArch].Artifacts["metal"].Formats["iso"].Disk
```

Comment:
Keep in mind this is example code :smile:

Reply:
Then we should make sure it doesn't have nil dereferences. :boom:

Reply:
There are nil values, not nil dereferences.   (Man I just want a time machine to go back and have Rust earlier so we'd have a nice `Option<T>` here probably...)

Dunno.  Perhaps what we want is a high level `FindArtifactDisk` API?  I actually ended up writing that over here https://github.com/openshift/installer/blob/master/pkg/rhcos/builds.go#L65
so it would make sense to lower here - basically check for nil values along the way?

Reply:
Er no the "metal" case is the case we actually can't do that.

Reply:
https://github.com/coreos/stream-metadata-go/pull/19

Reply:
Do you consider this blocking?

Reply:
> There are nil values, not nil dereferences.

Whoops, you're right, sorry.  Objection withdrawn.